geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommandQuery.java

```diff
@@ -262,9 +279,10 @@ protected boolean processQueryUsingParams(Message msg, Query query, String query
       checkForInterrupt(servConn, e);
       // Otherwise, write a query response and continue
       // Check if query got canceled from QueryMonitor.
-      DefaultQuery defaultQuery = (DefaultQuery) query;
-      if ((defaultQuery).isCanceled()) {
-        e = new QueryException(defaultQuery.getQueryCanceledException().getMessage(),
+      if (e instanceof QueryExecutionLowMemoryException
+          || e instanceof QueryExecutionTimeoutException
+          || e instanceof QueryExecutionCanceledException) {
+        e = new QueryException(e.getMessage(),
```

Comment:
shouldn't we call the single-arg constructor here: `new QueryException(e)`?

I think we only want this two-arg version if we are changing the message (which we are not)

Reply:
Hrmmm I did not actually change the exception construction itself from how it was done before.  Looking at GemFireCheckedException (the base class), it looks like the message would be lost if we didn't use the 2 arg constructor, which I agree is not how you would expect it to work, but it is.  Therefore I think I'll leave it as is.